docs/proposal/internal-ipv6-handling-rollback.md

```diff
@@ -0,0 +1,74 @@
+<!-- clone me -->
+```yaml
+---
+title: "Proposal: Rollback IPv6 internal Node IP exposure"
+version: if applicable
+authors: https://github.com/ivankatliarchuk
+creation-date: 2025-01-01
+status: proposal
+---
+```
+
+# Introduce Feature Flag for IPv6 Internal Node IP Handling in ''external-dns'' and Change the behavior
+
+## Summary
+
+This proposal aims to introduce a feature flag in 'external-dns' to control the handling of IPv6 internal node IPs. In the current version, the feature flag will default to the existing behavior. In the next `minor` or `minor+N` version, the default behavior will be reversed, encouraging users to adopt the new behavior while providing a transition period.
+
+## Motivation
+
+The discussion in [issue#4566](https://github.com/kubernetes-sigs/external-dns/issues/4566) and the subsequent [pr#4574](https://github.com/kubernetes-sigs/external-dns/pull/4574) and [pr#4808](https://github.com/kubernetes-sigs/external-dns/pull/4808) highlighted concerns regarding the treatment of IPv6 internal node IPs. To address these concerns without causing immediate disruption, a feature flag will allow users to opt-out the current behavior, providing flexibility during the transition.
+
+## Proposal
+
+- ***Introduce Feature Flag***
+   -  Add a feature flag, e.g., `--expose-internal-ipv6=true`, to control the handling of IPv6 internal node IPs.
+   -  In the current version, this flag will default to `true`, maintaining the existing behavior.
+
+- ***Flip Default Behavior in Next Minor Version***
+    -  In the subsequent minor release, change the default value of `--expose-internal-ipv6` to `false`, adopting the new behavior by default.
+    -  Users can still override this behavior by explicitly setting the flag as needed.
+
+- ***Propose/Add an annotation for this specific use case***
+    -  Provide support for `external-dns.alpha.kubernetes.io/expose-internal-ipv6` in follow-up releases.
+    -  Managing dual annotation and flag may introduce complexity.
```

Comment:
```suggestion
```

FTM, I think I have a strong opinion against this annotation with its induced complexity.
My head hurts just trying to imagine a use case requiring this level of complexity.
Please let me know if you have one, otherwise I suggest to avoid this complexity, at least FTM.

Reply:
Moved to non-goal section. Basically it was based on issue and from discussion. I have no strong opinion on that stuff.

Reply:
Just for reference, when it was proposed by maintainer
- https://github.com/kubernetes-sigs/external-dns/pull/4574#issuecomment-2284227526

But reading more thread, could not be a great idea.